RubberduckTests/Refactoring/RenameTests.cs

```diff
@@ -21,6 +21,74 @@ namespace RubberduckTests.Refactoring
     public class RenameTests : VbeTestBase
     {
         [TestMethod]
+        public void RenameRefactoring_RenameSub_Issue2727()
+        {
+            //Input
+            string inputCode = GetIssue2727ExampleCode();
+
+            Selection selection = Select2727Variable();
+
+            //New name provided by the user - no conflicts
+            var userEnteredName = "Value";
+
+            //Expectation
+            //Expecation is that the messageBox.Show() is not invoked
+
+            //Arrange
+            var builder = new MockVbeBuilder();
+            IVBComponent component;
+            var vbe = builder.BuildFromSingleStandardModule(inputCode, out component, selection);
+            var project = vbe.Object.VBProjects[0];
+            var module = project.VBComponents[0].CodeModule;
+            var mockHost = new Mock<IHostApplication>();
+            mockHost.SetupAllProperties();
+            var parser = MockParser.Create(vbe.Object, new RubberduckParserState(vbe.Object));
+
+            parser.Parse(new CancellationTokenSource());
+            if (parser.State.Status >= ParserState.Error) { Assert.Inconclusive("Parser Error"); }
+
+            var qualifiedSelection = new QualifiedSelection(new QualifiedModuleName(component), selection);
+
+            var msgbox = new Mock<IMessageBox>();
+            msgbox.Setup(m => m.Show(It.IsAny<string>(), It.IsAny<string>(), MessageBoxButtons.YesNo, It.IsAny<MessageBoxIcon>()))
```

Comment:
You've set up the mock - if the test means to verify that msgbox.show was never invoked, the `msgbox.Verify()` condition failing would be enough to fail the test- no need to `Assert` if you `Verify` ;-)